### PR TITLE
Add automatically default value for deployment PVC

### LINKED ISSUE #9585: UI workload creation with storage should pick the default storage class
**Setup**
- Rancher version: 2.7.5
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**
 Creating a workload from the UI with storage should default to the default storage class. 

**To Reproduce**

1. Create a default StorageClass in a cluster
2. For the same cluster, select Workloads > Deployments > Create from the UI
3. Select Pod > Storage > Add Volume > Create PVC 

**Result**
 The storage class dropdown is empty
 
**Expected Result**
The storage class field should default to the default storage class

**Screenshots**
image
image

**Additional context**

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #9585

### Occurred changes and/or fixed issues
Default value is now automatically assigned after fetching the data.

### Technical notes summary
Unit test has been created but it's not possible to use it till we do not integrate `fetch()` in the test suite.

### Screenshot/Video

https://github.com/rancher/dashboard/assets/5009481/70b29e37-a15b-456d-a786-beb2d990c7a3

### CODE REVIEW

File: shell/edit/workload/storage/persistentVolumeClaim/persistentvolumeclaim.vue [hunk 1]
```diff
@@ -91,12 +100,11 @@ export default {
   },
 
   watch: {
-    createPV(neu, old) {
+    isCreatePV(neu, old) {
```
Comment (torchiaf) [inline shell/edit/workload/storage/persistentVolumeClaim/persistentvolumeclaim.vue:103]: `old` is never used, it can be removed.

**Review by torchiaf — CHANGES_REQUESTED** [review]
- The e2e test need to be verified, we could probably just skip the e2e tests and address them in a new issue.
- PVC creation looks good when creating a new Deployment from scratch, but it seems to be still broken when edit a deployment and add a new PVC:

image

I think this case should be covered by this fix.

**Comment by cnotv** [comment]
The E2E unfortunately is failing for everyone 😬

**Comment by cnotv** [comment]
> * PVC creation looks good when creating a new Deployment from scratch, but it seems to be still broken when edit a deployment and add a new PVC:

> I think this case should be covered by this fix.

Totally agree and nice catch. Not sure why I was restricting it to creation only.

**Review by torchiaf — APPROVED** [review]
Approved once the e2e tests complete with success.

**Pull request merged**